joomla/joomla-cms
# [4.0]Kill hathor support code in list models

Pull Request for Issue # .

### Summary of Changes
For backend component model classes which extends JModelList, the model states are populated automatically in populateState method of JModelList.

However, we still had to set model states manually in every model lists class to support hathor template. Now, that template is removed in Joomla 4, we can remove these code

### Testing Instructions
There are some components affected by this change. Since change is the same for all components, you can pick one component to test (For example.com_content).

1. Install Joomla 4 (with sample data for easier testing)
2. Access to Articles management screen
3. Use search tool (enter keyword to search, select a category...), see the list of articles changed and filtered properly
4. Navigate to a different component (for example, cpanel of administrator area)
5. Access to Articles management screen screen again, make sure the data you entered in search tool (keyword, selected category...) still keep it's previous value.
